File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    const lenT = t.length;
    const lenP = p.length;
    const bigP = BigInt(p);
    let count = 0;
    
    for (let i = 0; i <= lenT - lenP; i++) {
        const str = BigInt(t.slice(i, i + lenP));
        
        if (str <= bigP) count++;
    }
    
    return count;
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    int_p = int(p)
    count = 0
    lt = len(t)
    lp = len(p)
    
    for i in range(lt-(lp-1)):
        n = int(t[i:i+lp])
        count += 1 if n <= int_p else 0
        
    return count